src/lib/ui/use-current-time.ts

```diff
@@ -0,0 +1,17 @@
+import { DateTime } from 'luxon'
+import { useCallback, useEffect, useState } from 'react'
+
+import { useInterval } from 'lib/ui/use-interval.js'
+
+export function useCurrentTime(): DateTime | null {
+  const [date, setDate] = useState<DateTime | null>(null)
+
+  const update = useCallback(() => {
+    setDate(DateTime.now())
+  }, [setDate])
+
+  useEffect(update, [update])
+  useInterval(update)
```

Comment:
Just curious, why to we need both vs just

```suggestion
  const update = useCallback(() => {
    setDate(DateTime.now())
  }, [setDate])

  useInterval(update)
```

Reply:
good point